### DISCUSSION TITLE
poetry cannot find distutils module

### DISCUSSION THREAD

**Discussion by Giuliopime** [python-poetry/poetry#8924] [Q&A]
The issue occurs when running `poetry self add poetry-plugin-up`which throws the following error:
```
No module named 'distutils'
```
I'm using python version 3.10.9 installed via pyenv, running `poetry env info`results in:
```Virtualenv
Python:         3.10.9
Implementation: CPython
Path:           /Users/j/Library/Caches/pypoetry/virtualenvs/account-3TXw94AP-py3.10
Executable:     /Users/j/Library/Caches/pypoetry/virtualenvs/account-3TXw94AP-py3.10/bin/python
Valid:          True
```
And distutils is located at `/Users/j/.pyenv/versions/3.10.9/lib/python3.10/distutils` yet poetry can't seem to find it?

Am I missing something? Looked everywhere online or in the repo issues but can't find the fix (I'm on mac)

**Comment by Giuliopime** [comment]
using `pipx inject poetry poetry-plugin-up` instead worked, still would like to understand why the other method didn't

**Reply by dimbleby** [reply]
poetry env info tells you about your project's environment, it has nothing to do with the environment in which poetry itself is installed

**Reply by Giuliopime** [reply]
How do I specify that? Reading through the docs it seemed that way
